src/util-proto-name.c

```diff
@@ -60,10 +61,9 @@ void SCProtoNameInit()
             if (proto_ch == NULL)
                 continue;
 
-            int proto = atoi(proto_ch);
-            if (proto >= 255)
+            uint8_t proto;
+            if (StringParseU8RangeCheck(&proto, 10, 0, (const char *)proto_ch, 0, 255) < 0)
```

Comment:
I suppose the range check is useless here, as every value a u8 can contain is valid

Reply:
The range should be useful here.

But `StringParseU8RangeCheck` should do it before doing the cast ie
First
```
    if (*res < min || *res > max) {
        return -1;
    }
```
Then 
`    *res = (uint8_t)u64;`

Reply:
Should the regular `StringParseU8` validate that the value fits the type as well?

Reply:
I would expect so

Reply:
@inashivb can you double check this?